sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -306,9 +308,27 @@ private static Item create(ItemSpec<?> spec, Path path) {
           spec.getValue(), spec.getShortValue(), spec.getLabel(), spec.getLinkUrl());
     }
 
+    /** <b>For internal use only. No backwards compatibility guarantees.</b> */
+    @Internal
+    public static Item raw(
+        Path path, String key, String value, String shortValue, String label, String linkUrl) {
+      checkNotNull(path, "path cannot be null");
+      checkNotNull(key, "key cannot be null");
+      return new AutoValue_DisplayData_Item(
+          path, Object.class, key, Type.STRING, value, shortValue, label, linkUrl);
+    }
+
     @Override
     public String toString() {
-      return String.format("%s%s:%s=%s", getPath(), getNamespace().getName(), getKey(), getValue());
+      return MoreObjects.toStringHelper(this)
```

Comment:
Does this `toString()` matter? The trouble is that when a test failed, the things printed here did not differ so I had to add the rest of the fields to find out what the failure was. (it turned out to be that `null` in a proto is defaulted to empty string... is that a problem?)